client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/provisioning/index.tsx

```diff
@@ -0,0 +1,56 @@
+import { Spinner } from '@wordpress/components';
+import { translate } from 'i18n-calypso';
+import { FC } from 'react';
+import './style.scss';
+
+type Status = 'idle' | 'pending' | 'success' | 'error';
+
+interface ProvisioningProps {
+	status: {
+		siteTransfer: Status;
+		pluginInstallation: Status;
+		migrationKey: Status;
+	};
+}
+
+export const Provisioning: FC< ProvisioningProps > = ( { status } ) => {
+	const {
+		siteTransfer: siteTransferStatus,
+		pluginInstallation: pluginInstallationStatus,
+		migrationKey: migrationKeyStatus,
+	} = status;
+
+	const actions = [
+		{ status: siteTransferStatus, text: translate( 'Provisioning your new site' ) },
+		{ status: pluginInstallationStatus, text: translate( 'Installing the required plugins' ) },
+		{ status: migrationKeyStatus, text: translate( 'Getting the migration key' ) },
+	];
+
+	const currentActionIndex = actions.findIndex( ( action ) => action.status !== 'success' );
+	const currentAction = actions[ currentActionIndex ];
+	if ( ! currentAction || currentAction.status === 'error' ) {
```

Comment:
The component is not handling errors in this PR. To test the component, you may have to comment out the error status check.